storage-s3-dynamodb/src/main/java/io/delta/storage/RetryableCloseableIterator.java

```diff
@@ -0,0 +1,147 @@
+package io.delta.storage;
+
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import org.apache.hadoop.fs.s3a.RemoteFileChangedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Iterates over the given iterator supplied in the constructor.
```

Comment:
you are not given an iterator. you are given a iterator generator. big difference...clarify.